src/query/atomWithQuery.ts

```diff
@@ -98,16 +98,17 @@ export function atomWithQuery<
         const unsubscribe = observer.subscribe(listener)
         return unsubscribe
       }
-      return { dataAtom, options }
+      return { dataAtom, observer, options }
     },
     (get, set, action: AtomWithQueryAction) => {
       switch (action.type) {
         case 'refetch': {
-          const { dataAtom, options } = get(queryDataAtom)
+          const { dataAtom, observer, options } = get(queryDataAtom)
           set(dataAtom, new Promise<TData>(() => {})) // infinite pending
-          const queryClient = get(getQueryClientAtom)
-          queryClient.getQueryCache().find(options.queryKey)?.reset()
-          const p: Promise<void> = queryClient.refetchQueries(options.queryKey)
+          const wrappedPromise = async () => {
```

Comment:
Please be reminded that in jotai code, there's no `async` anywhere. 🤯 
(well, you should know because you did all bundling tooling.)

If `const p: Promise<void> = ...` hack doesn't work, please use `Promise.resolve(...)`.